modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy

```diff
@@ -368,6 +368,35 @@ def checkoutFromGitHubWithSSH(String repository, Map options = [:]) {
   }
 }
 
+/**
+ * Checkout a dependent repo.
+ * This function acts as a wrapper around checkoutFromGitHubWithSSH with
+ * options tailored towards the needs of a secondary repo cloned as part of a
+ * pipeline job
+ *
+ * It can accept an optional closure that is run within the directory that has
+ * been cloned
+ */
+def checkoutDependent(String repository, options = [:], Closure closure = null) {
+  def defaultOptions = [
+    branch: "master",
+    changelog: false,
+    directory: "tmp/${repository}",
+    poll: false
+  ]
+  options = defaultOptions << options
+
+  stage("Cloning ${repository}") {
+    checkoutFromGitHubWithSSH(repository, options)
+  }
+
+  if (closure) {
```

Comment:
It might be nice to wrap this in a stage (maybe using a name configured in `options`) so the pipeline output is clearer. But happy for you to leave it like this for now - we can revisit this if we do add any long processes which use the dependent repo.

Reply:
Yeah I wrestled with that for a little bit as I'd agree it'd be better in a stage. The auto stage naming approach was so vague to be a bit pointless "${repository} tasks" sort of thing. Whereas passing in a stage is probably less intuitive than writing your own within the closure. 

Happy to evaluate after a bit though if it causes problem.